src/secret.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2024  Andrew Bower
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * \file secret.c
+ * Routines to manage access to credential store.
+ *
+ * \ingroup secret
+ * \{
+ **/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <err.h>
+#include <errno.h>
+#include <locale.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <termios.h>
+#include "gettext.h"
+#include "defs.h"
+#include "mem.h"
+#include "options.h"
+#include "prompt.h"
+#include "secret.h"
+
+#if HAVE_LIBSECRET
+#include <libsecret/secret.h>
+
+#define MCDS_NAMESPACE "com.github.t-brown.mcds"
```

Comment:
I'm not crash hot on `#define`s and prefer `static const char`. However, this does mean you can concat them easily.

Reply:
Fair enough! I don't think there's a particular need not just to put the full identifier in one string constant - do you want to do that change?

Reply:
Also, you may have a better schema id in mind - I just constructed one from this repo by default!